packages/react-core/src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Menu/menu';
+import { css } from '@patternfly/react-styles';
+import { getOUIAProps, OUIAProps, getDefaultOUIAId } from '../../helpers';
+import { SearchInput } from '../SearchInput';
+import { Divider } from '../Divider';
+import { MenuContext } from './MenuContext';
+
+export interface MenuProps
+  extends Omit<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, 'ref' | 'onSelect'>,
+    OUIAProps {
+  /** Anything that can be rendered inside of the Menu */
+  children?: React.ReactNode;
+  /** Additional classes added to the container */
+  className?: string;
+  /** Callback for updating when item selection changes */
+  onSelect?: (event: React.MouseEvent, itemId: any) => void;
+  /** Callback called when an MenuItems's favorite button is clicked */
+  onFavorite?: (event: React.MouseEvent, itemId: any) => void;
+  /** A callback for when the input value changes. */
+  onSearchInputChange?: (
+    event: React.FormEvent<HTMLInputElement> | React.SyntheticEvent<HTMLButtonElement>,
+    value: string
+  ) => void;
+  /** Accessibility label */
+  'aria-label'?: string;
+  /** Indicates menu type */
+  variant?: 'default' | 'flyout' | 'singleSelect' | 'multiSelect';
```

Comment:
is this needed? I only see `variant="flyout"` used in the examples. If it's not needed maybe it's better to remove it and add a `isFlyout` boolean prop onstead